enhancements/manifestlist/manifestlist-support.md

```diff
@@ -0,0 +1,403 @@
+      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+      "platform": {
+        "architecture": "arm",
+        "os": "linux",
+        "variant": "v5"
+      },
+      "size": 1362
+    }
+  ],
+}
+```
+
+This enhancement proposal designs a plan to add support for this new type of manifest in
+OpenShift.
+
+## Motivation
+
+When running on multi-architecture clusters it makes sense to be able to allow users to work with
+images that support different architectures as well. By adding support for manifest lists we are
+also adding support for hosting these multi-architecture images in OpenShift's internal registry.
+
+### Goals
+
+- Allow users to import manifest lists using the `oc` client.
+- Allow users to execute pull-through of manifest lists.
+- Make image pruner aware of manifest lists.
+- Allow users to push manifest lists into the integrated registry.
+- Improve the `oc describe` command to present manifest list info.
+
+### Non-Goals
+
+- Modify OpenShift's web console to support manifest lists.
+- Designs OpenShift multi-architecture builds.
+- Provide facilities to build manifest lists.
+
+## Proposal
+
+The proposed approach involves allowing one Image struct to refer to multiple other images by
+their hashes. In such a scenario, an Image `A` would represent the manifest list containing among
+its properties a slice of other Images `B` and `C`, each of them being a Manifest for an image of
+a different platform.
+
+In other words, when seeing from an ImageStream point of view we would have a mapping that looks
+like this:
+
+```text
+ImageStream X -> Image A -> Image B
+                            Image C
+                            Image D
+
+```
+
+Here `Image A` would contain references to `B`, `C`, and `D`. As a manifest list is simply a
+reference to other manifests they do not have any extra information such as "environment
+variables", "entry point" or "open ports" among its fields, this extra information, therefore,
+still belongs to the "child images" (`B` and `C` for instance). Image A would serve as a means
+to group other Images.
+
+### User Stories
+
+#### Support for manifest list import
+
+As an OpenShift user
+I want to be able to import manifest lists
+So that I can refer to images supporting multiple platforms
+
+#### Support for manifest list pull-through
+
+As an OpenShift user
+I want to be able to pull through manifest lists
+So that I can cache multi-architecture images within my cluster
+
+#### Support for pushing manifest lists to OpenShift's image registry
+
+As an OpenShift user
+I want to be able to push manifest lists to Openshift's internal registry
+So that I can publish my multi-architecture images
+
+#### Support for manifest lists during prune
+
+As an OpenShift user
+I want to be able to automatically clean unused manifest lists
+So that I can keep my storage resource usage under control
```

Comment:
need more detail on how prune will work.  If any image within the manifestlist is in use, will the full manifestlist be kept around?  Or only if the manifestlist itself is referenced the manifestlist will be kept around?

or something in between?  (if 2 of 3 images in the manifestlist are in use, the manifestlist is kept around but the unused image is removed from the manifestlist)

Reply:
I think that the best way of dealing with this would be to maintain the whole struct consistent by not removing anything that is part of the manifest list.  I have added a [section](https://github.com/openshift/enhancements/pull/635/files#diff-b4a80ec15b1d903d24929518bc457be2aea5a1ba81d43eccd586314da952a168R363) to the proposal specifying this, PTAL.

/cc @dmage you may have some input on this regard.

Reply:
keeping the entire manifestlist intact if any image it contains is ref'd makes sense to me.